### ISSUE TITLE
possible compatibility problem with Matplotlib-3.5.0 ?

### ISSUE THREAD

**Issue by stonebig** [networkx/networkx#5167]
doing my usual global test on coming matplotlib-3.5.0rc1, or matplotlib-3.5.0.dev2445+gb09aad279b-cp39-cp39-win_amd64.whl of Matplotlib-master CI of october 30th, I notice a problem of display.

````
%matplotlib inline
import matplotlib.pyplot as plt
import networkx as nx

g = nx.random_lobster(16, 0.5 , 0.5/16)
nx.draw(g)
plt.title('lobster')
plt.show()
```` 

### Current Behavior
![image](https://user-images.githubusercontent.com/4312421/139746028-a5416f89-224d-44ba-b35f-2cf4b4d96674.png)
### Expected Behavior
when using matplotlib-3.4.3, I have no problem
![image](https://user-images.githubusercontent.com/4312421/139746070-ab486638-035e-489b-a87d-6ecd0b18247a.png)

### Steps to Reproduce
install matplotlib-3.5.0rc1 (or current matplotlib-master ?)

run the example on jupyterlab or jupyter notebook.

### Environment
 Python version: 3.9.7, on Windows 10, on pip
NetworkX version: 2.6.3 

### Additional context
problem also reported to Matplotlib, as I'm not sure of the root cause https://github.com/matplotlib/matplotlib/issues/21517

**Comment by rossbar** [comment]
Thanks for the report @stonebig , I can reproduce this as well with matplotlib 3.5.0-rc1.

Just to note, in the first image in the OP, if you zoom into the lower left corner, you will see the nodes and the edges (the tiny black dot in the image) are both drawn, but in different locations and (potentially) with different scales. I've just run the NX docs build with mpl 3.5-rc1 to look at the sphinx gallery and it looks like this is a pervasive problem, particularly with `LineCollection` which is used by default to represent undirected edges. Directed edges (represented with FancyArrowPatches) still seem to be fine.

For anyone with a little extra time looking to help track this down, one useful activity would be to run `git bisect` matplotlib to identify the commit that's causing the behavior change.

**Comment by rossbar** [comment]
I went ahead and did the bisection which identified matplotlib/matplotlib@1f4708b310 as the source of the behavior change. I've reported there and we'll see whether this is the intended behavior, in which case we'll have to figure out how to update `nx_pylab` accordingly.

**Comment by rossbar** [comment]
From the discussion in matplotlib/matplotlib#21517, this looks like it needs a fix on the networkx side - fortunately, it is quite minor.

However we may also want to consider another NetworkX 2.6 patch to either backport the fix or pin matplotlib < 3.5 so that users who happen to get NetworkX 2.6 with Matplotlib 3.5 aren't bit by this.
